lib/mnesia/src/mnesia.erl

```diff
@@ -2610,31 +2879,39 @@ set_master_nodes(Tab, Nodes) ->
 
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 %% Misc admin
-
+-spec dump_log() -> 'dumped'.
 dump_log() ->
     mnesia_controller:sync_dump_log(user).
 
+-spec sync_log() -> 'ok' | {'error', Reason::term()}.
 sync_log() ->
     mnesia_monitor:sync_log(latest_log).
 
+-spec subscribe(What) -> {'ok', node()} | {'error', Reason::term()} when
+      What :: 'system' | 'activity' | {'table', table(), 'simple' | 'detailed'}.
```

Comment:
`'simple' | 'detailed'` should be a type, both so that it has a descriptive name and avoid code duplication.